### PR TITLE
Enable fp64 conversion emulation conditionally

### PR DESCRIPTION
**Pull request by Chao1Han**

### CODE REVIEW

File: cmake/BuildFlags.cmake [hunk 1]
```diff
@@ -64,8 +64,16 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "MSVC"
     set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -D_GLIBCXX_USE_CXX11_ABI=${GLIBCXX_USE_CXX11_ABI})
   endif()
 
-  set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -fsycl-fp64-conv-emu)
-  
+  if(SYCL_COMPILER_VERSION)
+    set(MIN_DATE_VERSION "20240604")
+    if(SYCL_COMPILER_VERSION GREATER_EQUAL MIN_DATE_VERSION)
+      message(STATUS "Enable Double Emulation Feature")
```
Comment (fengyuan14) [inline cmake/BuildFlags.cmake:70]: Remove the message if workable.

File: cmake/BuildFlags.cmake [hunk 2]
```diff
@@ -64,8 +64,16 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "MSVC"
...
+      set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -fsycl-fp64-conv-emu)
+    else()
+      message(STATUS "SYCL Compiler Version ${SYCL_COMPILER_VERSION} is less than ${MIN_DATE_VERSION}. Disable Double Emulation Feature")
```
Comment (fengyuan14) [inline cmake/BuildFlags.cmake:73]: I think `Warning` is better. And please enhance the log,
`"SYCL Compiler Version ${SYCL_COMPILER_VERSION} is less than ${MIN_DATE_VERSION}. Disable FP64 conversion emulation. There might be unknown behavior in some platforms that don't support FP64."`

Comment (Chao1Han) [reply cmake/BuildFlags.cmake]: Done

File: cmake/BuildFlags.cmake [hunk 3]
```diff
@@ -1,5 +1,30 @@
 # Setup building flags for SYCL device and host codes.
 
+function(CHECK_SYCL_FLAG FLAG VARIABLE_NAME)
+set(TEMP_DIR "${CMAKE_BINARY_DIR}/temp")
+file(MAKE_DIRECTORY ${TEMP_DIR})
+set(TEST_SRC_FILE "${TEMP_DIR}/demo.cpp")
```
Comment (fengyuan14) [inline cmake/BuildFlags.cmake:6]: Can we name it as `check_options.cpp`

File: cmake/BuildFlags.cmake [hunk 4]
```diff
@@ -64,8 +89,15 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "MSVC"
     set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -D_GLIBCXX_USE_CXX11_ABI=${GLIBCXX_USE_CXX11_ABI})
   endif()
 
-  set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -fsycl-fp64-conv-emu)
-  
+  CHECK_SYCL_FLAG("-fsycl-fp64-conv-emu" SUPPORTS_FP64_FLAG)
+  if(SUPPORTS_FP64_FLAG)
```
Comment (fengyuan14) [inline cmake/BuildFlags.cmake:93]: `SUPPORTS_FP64_CONV_EMU`

File: cmake/BuildFlags.cmake [hunk 5]
```diff
@@ -64,8 +89,15 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "MSVC"
...
+    set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -fsycl-fp64-conv-emu)
+  else()
+    message(WARNING "On some platforms that don't support FP64, \
```
Comment (fengyuan14) [inline cmake/BuildFlags.cmake:96]: Please enhance the log.

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**